Add support for django.contrib.auth.forms.UsernameField

ajcerejeira opened the pull request:
`django.contrib.auth.forms` defines two form fields, [UsernameField](https://github.com/django/django/blob/f81032572107846922745b68d5b7191058fdd5f5/django/contrib/auth/forms.py#L72) and [ReadOnlyPasswordHashField](https://github.com/django/django/blob/f81032572107846922745b68d5b7191058fdd5f5/django/contrib/auth/forms.py#L63) that are used by django admin website.

At the moment `hypothesis` does not define a default strategy for `UsernameField` so this MR adds that support by reusing the same strategy for `CharField`.

Partial fix for #3417.  Incorporates and closes #3420, incorporates and closes #3421 - neither really need a separate release.

Zac-HD reviewed (commented):
Looks like a great start - thanks so much for contributing!

Remaining things: add yourself to the list in `AUTHORS.rst`, `ReadOnlyPasswordHashField` support, and adding a `hypothesis-python/RELEASE.rst`:

```rst
RELEASE_TYPE: minor

:func:`~hypothesis.extra.django.from_field` now supports
:class:`~django.contrib.auth.forms.UsernameField` and
:class:`~django.contrib.auth.forms.ReadOnlyPasswordHashField`
from ``django.contrib.auth.forms``.

Thanks to Afonso Silva for reporting and fixing :issue:`3417`.
```

ajcerejeira commented:
Thanks @Zac-HD, I've added myself to the authors list and added a release file.
About `ReadOnlyPasswordHashField`, this PR already includes support for that, i.e, this field was already working for hypothesis, I simply added a form test to ensure it anyway.

Zac-HD commented on hypothesis-python/RELEASE.rst:
```diff
@@ -0,0 +1,8 @@
+RELEASE_TYPE: minor
+
+:func:`~hypothesis.extra.django.from_field` now supports
+:class:`~django.contrib.auth.forms.UsernameField` and
+:class:`~django.contrib.auth.forms.ReadOnlyPasswordHashField`
+from ``django.contrib.auth.forms``.
+
+Thanks to Afonso Silva for reporting and fixing :issue:`3417`.
```
```suggestion
:class:`~django.contrib.auth.forms.UsernameField` from ``django.contrib.auth.forms``.

Thanks to Afonso Silva for reporting and working on :issue:`3417`.
```

Since we already supported the other field, and have some remaining things in the issue.

Zac-HD reviewed (commented):
Looks good to me!  Once we work out why this broke the docs build (and unbreak it), I'll merge 🎉 

And then we can take a look at the other points in #3417 😁

Zac-HD commented:
Thanks again @ajcerejeira!
